app/src/main/java/com/example/recipeapp/Utils/GalleryHandler.java

```diff
@@ -0,0 +1,79 @@
+package com.example.recipeapp.Utils;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.ImageDecoder;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.MediaStore;
+
+import androidx.activity.ComponentActivity;
+import androidx.activity.result.ActivityResult;
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.parse.ParseFile;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class GalleryHandler {
+    private static final String NAME = "myImage";
+    private static final String INTENT_TYPE = "image/*";
+    public ActivityResultLauncher<Intent> launcher;
+    ComponentActivity activity;
+
+    public GalleryHandler(AppCompatActivity activity, ActivityResultCallback<ActivityResult> callback) {
+        this.activity = activity;
```

Comment:
You should not hold activity inside a object this might cause Memory Leak here. 

ref - https://stackoverflow.com/questions/15486676/what-are-strong-pointers-and-weak-pointers